### PR TITLE
Feature: Add Random Delay to Interactions

### LINKED ISSUE #12: Random sleep time request
Greetings. I want to offer the ability to adjust the reaction time.
Something of

```
from random import gauss
def reaction_time():
    max = 0.33
    min = 0.15

    reaction = gauss(0.225, 0.6)
    if (reaction < min): return min
    if (reaction > max): return max
    return reaction

    ...
    sleep(reaction_time())
```

### LINKED ISSUE #15: Reel in randomization
Randomize the “real in” time instead of almost always instant. Is this possible already and I just can’t figure out how to do it? Great mod btw, nice and simple!

### PR DESCRIPTION
**Pull request by SemjonKerner**
Adds an option to have human-like/random jitter in delays. This can be turned on and off from the configs.

~~Mergeconflict with other PRs incoming.~~

Fixes #12, Fixes #15

### CODE REVIEW

File: fishy/engine/semifisher/fishing_event.py [hunk 1]
```diff
@@ -32,9 +34,16 @@ class FishEvent:
     uid = None
     sound = False
 
+def _fishing_sleep(waittime, lower_limit = 16, upper_limit = 58):
```
Comment (adsau59) [inline fishy/engine/semifisher/fishing_event.py:37]: usually time in python is considered in seconds, if you want millisecs try to name ur variable with `_ms` in the end or mention it in doc string to improve code readability

Comment (adsau59) [inline fishy/engine/semifisher/fishing_event.py:37]: 16-58 ms seem too less, try to find out the max time we can delay reeling in the fish, then add an offset just so the fish doesn't escapes sometimes, use those values (I'm not sure if you are already doing that 😅 ignore this if you are)

**Review by adsau59 — CHANGES_REQUESTED** [review]
thanks for the contribution, just requesting couple of small changes...

File: fishy/engine/semifisher/fishing_event.py (hunk 1 above)
Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py:37]: Okay. This is just to avoid hickups with randrange. Will call it *_ms then.

Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py:37]: I can try that, will take a moment though. But I will only trial and error this by hand, so the values won't be to sharp. From my testings I think 16-58ms are very human like. Thats about the time it takes me to click, as long as I am attentive.

Comment (adsau59) [reply fishy/engine/semifisher/fishing_event.py:37]: yeah but imagine, people wont be that attentive when they are fishing for hours 😉

File: fishy/engine/semifisher/fishing_event.py [hunk 2]
```diff
@@ -34,11 +34,12 @@ class FishEvent:
     uid = None
     sound = False
 
-def _fishing_sleep(waittime, lower_limit = 16, upper_limit = 58):
+def _fishing_sleep(waittime, lower_limit_ms = 16, upper_limit_ms = 2500):
     reaction = 0.0
-    if (FishEvent.jitter):
-        reaction = float( random.randrange(lower_limit, upper_limit) )/100.0
-    time.sleep(waittime+reaction)
+    if (FishEvent.jitter and (upper_limit_ms > lower_limit_ms)):
```
Comment (adsau59) [inline fishy/engine/semifisher/fishing_event.py:39]: no need for brackets

File: fishy/engine/semifisher/fishing_event.py [hunk 3]
```diff
@@ -34,11 +34,12 @@ class FishEvent:
...
+        reaction = float( random.randrange(lower_limit_ms, upper_limit_ms) )/1000.0
+    max_wait_t = waittime+reaction if (waittime+reaction <= 2.5) else 2.5
```
Comment (adsau59) [inline fishy/engine/semifisher/fishing_event.py:41]: no need for brackets

**Review by adsau59 — CHANGES_REQUESTED** [review]

File: fishy/engine/semifisher/fishing_event.py (hunk 2 above)
Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py:39]: I make them for readability and I like them in this and the other case.

Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py:39]: It's saying, look this part is a logical unit.

File: fishy/engine/semifisher/fishing_event.py (hunk 3 above)
Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py:41]: see above

File: fishy/engine/semifisher/fishing_event.py (hunk 2 above)
Comment (adsau59) [reply fishy/engine/semifisher/fishing_event.py:39]: actually its pythonic to not use brackets until there are alot of `and` and `or` conditions u need to group up... for a simple `and` between two bools, its better if brackets doesn't clutter up the code

**Review by adsau59 — CHANGES_REQUESTED** [review]

File: fishy/engine/semifisher/fishing_event.py (hunk 2 above)
Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py:39]: Okay If you want it that way. In my opinion having a comparison makes it complicated to see on first glance.

Comment (SemjonKerner) [reply fishy/engine/semifisher/fishing_event.py]: You want the outer brackets gone too, right?

Comment (adsau59) [reply fishy/engine/semifisher/fishing_event.py]: > In my opinion having a comparison makes it complicated to see on first glance.

I understand, I too used to have the same opinion coming from java to python, but then years of using pycharms it started to feel like brackets clutter up the code for no reason

> You want the outer brackets gone too, right?

Yes, and I think you did it, so accepting the pr

**Comment by adsau59** [comment]
Thanks for the contribution 😄

**Pull request merged**